R/git.R

```diff
@@ -154,31 +154,51 @@ git_create_sample_log <- function(folder_path="/tmp"){
   git_repo <- file.path(folder_path,'.git')
 
   # git --git-dir sample/.git --work-tree sample add hello.R
+  #' @param git_repo The git repo path
+  #' @param folder_path The worktree path
+  #' @param file The file we want to add
+  #' @export
+  git_add <- function(git_repo, folder_path, file) {
+
   error <- system2('git',
                    args = c('--git-dir',
                             git_repo,
                             '--work-tree',
                             folder_path,
                             'add',
-                            '.'),
+                            file),
                    stdout = TRUE,
                    stderr = FALSE)
+  }
+
+
+  # calling functions
+  git_add(git_repo, folder_path, "hello.R")
+  git_commit(git_repo, folder_path, "This is a commit msg.", "John Smith", "test@email.com")
```

Comment:
This is not a script file. It should not execute code outside function definitions.